clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/manifest/KubernetesPatchManifestOperation.java

```diff
@@ -64,15 +66,26 @@ public OperationResult operate(List _unused) {
     updateStatus("Finding patch handler for " + objToPatch + "...");
     KubernetesHandler patchHandler = findPatchHandler(objToPatch);
 
-    updateStatus("Swapping out artifacts in " + objToPatch + " from context...");
-    ReplaceResult replaceResult = replaceArtifacts(objToPatch, patchHandler);
-
-    updateStatus("Submitting manifest " + description.getManifestName() + " to Kubernetes master...");
     OperationResult result = new OperationResult();
-    result.merge(patchHandler.patch(credentials, objToPatch.getNamespace(), objToPatch.getName(),
-      description.getOptions(), replaceResult.getManifest()));
 
-    result.getBoundArtifacts().addAll(replaceResult.getBoundArtifacts());
+    KubernetesManifest manifestPatch = description.getPatchBody().getManifestPatch();
+    List<JsonPatch> jsonPatches = description.getPatchBody().getJsonPatches();
+
+    if (manifestPatch != null) {
```

Comment:
I would recommend checking the patch strategy in the incoming payload instead of checking which of these fields was defined.

Reply:
Makes sense, updated!